tests/st/bert/bert_for_pre_pretraining.py

```diff
@@ -0,0 +1,342 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""
+#################pre_train bert example on zh-wiki########################
+                    python run_pretrain.py
+"""
+
+import os
+import argparse
+import logging
+
+import mindspore.communication.management as D
+from mindspore.nn.wrap.loss_scale import DynamicLossScaleUpdateCell
+
+
+from tinyms import data as ds
+from tinyms import vision
+
+from tinyms import context
+from tinyms.model import Model
+from tinyms.context import ParallelMode
+from tinyms.callbacks import ModelCheckpoint, CheckpointConfig, TimeMonitor
+from tinyms.optimizers import AdamWeightDecay, Lamb, Momentum, THOR
+from tinyms import set_seed
+
+
+from tinyms.model.bert import BertNetworkWithLoss, BertTrainOneStepCell, BertTrainOneStepWithLossScaleCell, \
+                BertTrainAccumulationAllReduceEachWithLossScaleCell, \
+                BertTrainAccumulationAllReducePostWithLossScaleCell, \
+                BertTrainOneStepWithLossScaleCellForAdam
+
+from tinyms.optimizers import AdamWeightDecayForBert, AdamWeightDecayOp
+from config import cfg, bert_net_cfg
+from utils import LossCallBack, BertLearningRate
+
+
+_current_dir = os.path.dirname(os.path.realpath(__file__))
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+)
+logger = logging.getLogger(__name__)
+
+
+
+
+def create_bert_dataset(device_num=1, rank=0, do_shuffle="true", data_dir=None, schema_dir=None):
+    """create train dataset"""
+    # apply repeat operations
+    files = os.listdir(data_dir)
+    data_files = []
+    for file_name in files:
+        if "tfrecord" in file_name:
+            data_files.append(os.path.join(data_dir, file_name))
+    data_set = ds.TFRecordDataset(data_files, schema_dir if schema_dir != "" else None,
+                                  columns_list=["input_ids", "input_mask", "segment_ids", "next_sentence_labels",
+                                                "masked_lm_positions", "masked_lm_ids", "masked_lm_weights"],
+                                  shuffle=True if do_shuffle == "true" else False,
+                                  num_shards=device_num, shard_id=rank, shard_equal_rows=True)
+    ori_dataset_size = data_set.get_dataset_size()
+    print('origin dataset size: ', ori_dataset_size)
+    type_cast_op = vision.TypeCast(ts.int32)
```

Comment:
Could u create a `BertDatasetTransform` class in `vision` module and move these data augment operations into it?